openmeter/customer/event.go

```diff
@@ -0,0 +1,145 @@
+package customer
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/openmeterio/openmeter/openmeter/event/metadata"
+)
+
+const (
+	CustomerEventSubsystem  metadata.EventSubsystem = "customer"
+	CustomerCreateEventName metadata.EventName      = "customer.created"
+	CustomerUpdateEventName metadata.EventName      = "customer.updated"
+	CustomerDeleteEventName metadata.EventName      = "customer.deleted"
+)
+
+// NewCustomerCreateEvent creates a new customer create event
+func NewCustomerCreateEvent(customer Customer) CustomerCreateEvent {
+	return CustomerCreateEvent{
+		EventEntityMutationPayload: metadata.EventEntityMutationPayload[Customer]{
+			Entity:       metadata.EntityCustomer,
+			MutationType: metadata.EntityMutationTypeCreate,
+			New:          &customer,
+		},
+	}
+}
+
+// CustomerCreateEvent is an event that is emitted when a customer is created
+type CustomerCreateEvent struct {
+	metadata.EventEntityMutationPayload[Customer] `json:",inline"`
+}
+
+func (e CustomerCreateEvent) EventName() string {
+	return metadata.GetEventName(metadata.EventType{
+		Subsystem: CustomerEventSubsystem,
+		Name:      CustomerCreateEventName,
+		Version:   "v1",
+	})
+}
+
+func (e CustomerCreateEvent) EventMetadata() metadata.EventMetadata {
+	return metadata.EventMetadata{
+		ID:     e.New.ID,
+		Source: metadata.ComposeResourcePath(e.New.Namespace, metadata.EntityCustomer, e.New.ID),
+		Time:   e.New.CreatedAt,
```

Comment:
Is Subject optional?

Reply:
we should add `namespace` and `user`

Reply:
Subject is optional, but strongly encouraged. It should match the Source in this specific case. Subject will determine the partition alignment.

I only see value in `namespace` being added here if we would want to filter for it without having do deserialize the object. Otherwise, it's contained in the entity; that's how we use it. 

What does the user mean in this context? What does it hold for a service key? 

Furthermore events are also created from workers (invoice state change from billing worker), there you would not have an user attribute.